app/src/main/java/com/example/rwblickhan/webviatextserver/MainActivity.java

```diff
@@ -1,26 +1,34 @@
 package com.example.rwblickhan.webviatextserver;
 
+
 import android.os.Handler;
 import android.os.Message;
 import android.os.Looper;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-import java.util.HashMap;
-
 public class MainActivity extends AppCompatActivity {
 
-    private static final int RECEIVE_SMS = 1;
+    private static MainActivity inst;
```

Comment:
This could lead to unexpected behavior since the Activity lifecycle is so complicated... instead make a listener interface and have MainActivity implement it.